### ISSUE TITLE
Can't dump from ryujit when used as an alt jit

### ISSUE THREAD

**Issue by AndyAyersMS** [dotnet/coreclr#2699]
If you set up ryujit as an alt jit, it won't dump anything. In particular, in a debug build

```
complus_jitdump=*
```
dumps everything, but
```
complus_jitdump=*
complus_altjit=*
complus_altjitname=ryujit.dll
```
dumps nothing.

**Comment by mikedn** [comment]
I think that ryujit.dll is not actually an altjit, it's just a dll version of the normal jit. If you define `ALT_JIT` then dump works fine. See the `ALT_JIT` #ifdefs in [`compInitOptions`](https://github.com/dotnet/coreclr/blob/master/src/jit/compiler.cpp#L2024).

**Comment by AndyAyersMS** [comment]
Believe this was largely addressed by #3431.
